provisioning/device/samples/readme.md

```diff
@@ -0,0 +1,101 @@
+
+X509 attestation comes in two flavors: 
+
+1. Group Enrollment
+In this case, a single Certificate Authority certificate is uploaded to the Provisioning Service. Devices have access to certificates issued by this CA. All devices will be using the same policies during IoT Hub provisioning.
+
+The service requires proof of possession of the uploaded CA certificate private key. This can be achieved using the [Proof of possession for X509 Group Enrollment](..\..\service\samples\GroupCertificateVerificationSample) tool.
+
+Because Intermediate Authorities may have been issued by the uploaded CA, the application must present the full chain of certificates from the one used during authentication to the one uploaded to the service. E.g.: If `My CA Certificate` was uploaded to the service and `MyDevice1` is the device certificate, the entire chain must be available on the device: 
+`<"My CA Certificate", "My Intermediate 1", ..., "My Intermediate N", "MyDevice1>`.
+
+2. Individual Enrollment
+In this case, each device certificate gets its own certificate uploaded to the service.
+_Note:_ the device certificate can also be a signing certificate of the actual certificate used to authenticate with the service during provisioning.
+
+An example of specifying the authentication X509Certificate using a PKCS12 PFX password-protected file:
+
+```C#
+    using (var certificate = new X509Certificate2(s_certificateFileName, certificatePassword))
+    using (var security = new SecurityClientX509(certificate))
+    {
+        //...
+    }
+```
+
+The SDK provides an extension model [SecurityClientHsmX509](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/shared/Microsoft.Azure.Devices.Shared/SecurityClientHsmX509.cs) that allows hardware vendors to implement custom Hardware Security Modules that store the device certificates. On Windows, PKCS11 HSM devices are supported through the [Certificate Store](https://docs.microsoft.com/en-us/windows-hardware/drivers/install/certificate-stores).
+
+An example of implementation for this extension module is the [SecurityClientX509](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/shared/Microsoft.Azure.Devices.Shared/SecurityClientX509.cs) class.
+
+### Provisioning devices using TPM based attestation
+
+In the case of TPM attestation, a RegistrationID must be supplied by the application.
+
+The TPM attestation supports only Individual Enrollments. The [Endorsement Key](https://technet.microsoft.com/en-us/library/cc770443(v=ws.11).aspx) (EK) must be supplied to the service. During provisioning, the [Storage Root Key](https://technet.microsoft.com/en-us/library/cc753560(v=ws.11).aspx) (SRK) will also be used to ensure that TPM ownership changes result in devices provisioned to the correct owner.
+
+
+```C#
+    using (var security = new SecurityClientTpmSimulator(RegistrationId))
+    {
+        //...
```

Comment:
Is this complete?

Reply:
I only wanted to point out the single line describing which security type to use. The `//...` is the part that's not TPM specific.

Reply:
Ah. Maybe add a "your code here" to the comment?